trin-core/src/bin/generate-data.rs

```diff
@@ -0,0 +1,138 @@
+use structopt::StructOpt;
+use rocksdb::{Options, DB};
+use trin_core::utils::{get_data_dir};
+use trin_core::portalnet::storage::{PortalStorage, PortalStorageConfig, DistanceFunction};
+use rand::{distributions::Alphanumeric, Rng};
+use std::process::Command;
+use discv5::enr::NodeId;
+use sha3::{Digest, Sha3_256};
+use std::convert::TryInto;
+
+pub fn main() -> Result<(), Box<dyn std::error::Error>> {
+
+    let generator_config = GeneratorConfig::from_args();
+
+    if generator_config.overwrite {
+        let _ = Command::new("rm")
+                        .arg("-rf")
+                        .arg(get_data_dir())
+                        .output()
+                        .expect("Failed to overwrite DB.");
+    }
+
+    let mut db_opts = Options::default();
+    db_opts.create_if_missing(true);
+    let db = DB::open(&db_opts, get_data_dir()).expect("Failed to open RocksDB.");
+
+    let num_kilobytes = generator_config.kb;
+    let size_of_keys = 32;
+    let size_of_values = generator_config.value_size;
+
+    // For every 1 kb of data we store (key + value), RocksDB tends to grow by this many kb on disk...
+    // ...but this is a crude empirical estimation that works mainly with default value data size. 
+    // TODO: Use perf::memory_usage_stats to be more accurate with all data sizes.
+    let data_overhead = 1.1783;
+    let number_of_entries = ( (num_kilobytes * 1000) as f64 / (size_of_values) as f64 ) / data_overhead;
+    let number_of_entries = number_of_entries.round() as u32;
+
+    let storage_config = PortalStorageConfig {
+        storage_capacity_kb: (num_kilobytes / 4) as u64,
+        node_id: NodeId::random(),
+        distance_function: DistanceFunction::Xor,
+    };
+    let mut storage = PortalStorage::new(&storage_config, |key| {
+        sha256(&key)
+    }).unwrap();
+
+    for _ in 0..number_of_entries {
+        
+        let value = generate_random_value(size_of_values);
+        let key = generate_random_value(size_of_keys);
+
+        println!("{} -> {}", &key, &value);
+
+        if generator_config.portal_storage {
+            storage.store(&key, &value);
+        } else {
+            db.put(&key, &value).expect("Failed to write DB entry.");
+        }
+
+        println!();
+
+    } 
+
+    db.flush().expect("Failed to flush changes to DB.");
+
+    println!("Successfully saved {} key/value pairs to RocksDB.", number_of_entries);
+
+    Ok(())
+
+}
+
+fn generate_random_value(number_of_bytes: u32) -> String {
+
+    rand::thread_rng()
+        .sample_iter(&Alphanumeric)
+        .take(number_of_bytes as usize)
+        .map(char::from)
+        .collect()
+
+}
+
+// Placeholder content key -> content id conversion function
+fn sha256(key: &String) -> [u8; 32] {
```

Comment:
I was wondering why use an array of 32 bytes instead of U256 data type?

Reply:
Good question, I moved away from U256s internally because SQLite won't take them so I had to convert them anyway, but this is external facing so I'll make it a U256.

Are you returning U256 from the derivation functions?

Reply:
Gotcha. I'm not 100% sure yet what the derivation functions will return. Currently, I'm using the sha3 crate that you're using, and my function returns whatever their `finalize()` function returns

Reply:
ok, i think that's returning a GenericArray. if we do decide to return a [u8; 32], you can see how to convert from that to a [u8; 32] in the above sha256 function. 

Also, just to put this here: I'm not using a U256 because I can't figure out how to turn one back into bytes. Primitive Rust integer types have `.to_bytes()` functions, U256 (built with parity's uint library) does not seem to have anything like that. It does have the `::from()` function which allows going in the other direction (byte array -> U256) just fine. Any ideas welcome.

Reply:
@victor-wei126 I can convert U256 to what I'm working with, so let's continue using U256.

Reply:
@mrferris Sorry this slipped through the cracks somehow, but yeah I'd support sticking with `U256`